### DISCUSSION TITLE
questions in data preparation of ppp of non-IGS station

### DISCUSSION THREAD

**Discussion by FFFFFG-IGG** [groops-devs/groops#104] [Q&A] [answered]
Hello, I'd like to deploy ppp with non-IGS station, but antennaDefination of satellite and reciver are not convenient to get, why shouldn't us use atx file from IGS for both antenna correction. What's more, I do not know how to get accuracyDifination file and are confused with it use. Last but not least, stationInfo files are too hard to obtain, for groops can not get that file from observation file, and the document of groops say "stationInfo files can be obtained from station log file or create it through staionInfoCreate", latter is not a good choice when simultaneous process set of stations, and former is beyond my knowledge, what's worse, I'm confused to get station Log file for non-IGS stations. So my questions are: In the case of processing non-IGS stations, how to get antenna defination files and accuracyDefination files conveniently? And how to get stationInfo fast if I process lots of stations simultaneously?
THX in advance!

**Comment by tmayerguerr** [comment]
Hi,

> Hello, I'd like to deploy ppp with non-IGS station, but antennaDefination of satellite and reciver are not convenient to get, why shouldn't us use atx file from IGS for both antenna correction.

Maybe you can use [GnssAntex2AntennaDefinition](https://groops-devs.github.io/groops/html/GnssAntex2AntennaDefinition.html)

> What's more, I do not know how to get accuracyDifination file and are confused with it use. 

Please see the answer in [Antenna Accuracy for a LEO satellite #109](https://github.com/groops-devs/groops/discussions/109).

> Last but not least, stationInfo files are too hard to obtain, for groops can not get that file from observation file, and the document of groops say "stationInfo files can be obtained from station log file or create it through staionInfoCreate", latter is not a good choice when simultaneous process set of stations, and former is beyond my knowledge, what's worse, I'm confused to get station Log file for non-IGS stations. 

At some point, you need to provide the necessary information. The easiest way is still to create station log files (see https://files.igs.org/pub/station/log/ for examples) and use [GnssStationLog2StationInfo](https://groops-devs.github.io/groops/html/GnssStationLog2StationInfo.html).

regards
Torsten

**Reply by FFFFFG-IGG** [reply]
Thanks for your kind reply! However, there comes question: if the stations are non-IGS stations, and the antenna of those stations do not have corresponding pattern in atx file(such as some not well-known vendors), groops will stop and that stations' observations will be banned. Is there any ways to process such stations in groops?

**Comment by Waley04** [comment]
I also have a similar problem. For example, I completely failed to process some CORS observation data from UNAVCO.

**Comment by tmayerguerr** [comment, marked as answer]
To process a station, an antenna definition is required. This is searched in the list e.g. antennaDefinition_igs14.dat.

The antenna list is searched from the end. The first antenna is used, where the name, serial number and radome fits.
Empty elements always fit. With the script [groops.xml.txt](https://github.com/groops-devs/groops/files/9741100/groops.xml.txt) an additional antenna definition is added, which always fits (empty name, serial, radome). The offset and PCV is assumed to be 0 here. Of course the results might not be optimal.

regards
Torsten
